service/storage_service.cc

```diff
@@ -5064,6 +5064,63 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
     });
 }
 
+future<locator::load_stats> storage_service::load_stats_for_tablet_based_tables() {
+    auto holder = _async_gate.hold();
+
+    if (this_shard_id() != 0) {
+        // group0 is only set on shard 0.
+        co_return co_await container().invoke_on(0, [&] (auto& ss) {
+            return ss.load_stats_for_tablet_based_tables();
+        });
+    }
+
+    auto load_stats = co_await _db.map_reduce0([] (replica::database& db) mutable -> future<locator::load_stats> {
+        locator::load_stats load_stats{};
+        auto& tables_metadata = db.get_tables_metadata();
+
+        co_await tables_metadata.for_each_table_gently([&] (table_id id, lw_shared_ptr<replica::table> table) -> future<> {
+            if (!table->uses_tablets()) {
+                return make_ready_future<>();
+            }
+
+            auto erm = table->get_effective_replication_map();
```

Comment:
It's more accurate to take this outside the outer loop (and do the magic to transport it across shards). In the way that it's written, migrations  can cause double-accounting.